bundles/org.openhab.binding.airquality/src/main/java/org/openhab/binding/airquality/internal/api/ApiBridge.java

```diff
@@ -66,21 +66,21 @@ private String buildRequestURL(String key, int stationId, String location) {
      * @return an air quality data object mapping the JSON response
      * @throws AirQualityException
      */
-    public AirQualityData getData(int stationId, String location, int retryCounter) throws AirQualityException {
+    public AirQualityData getData(int stationId, String location) throws AirQualityException {
         String urlStr = buildRequestURL(apiKey, stationId, location);
         logger.debug("URL = {}", urlStr);
 
         try {
             String response = HttpUtil.executeUrl("GET", urlStr, null, null, null, REQUEST_TIMEOUT_MS);
             logger.debug("aqiResponse = {}", response);
             AirQualityResponse result = GSON.fromJson(response, AirQualityResponse.class);
-            if (result != null && result.getStatus() == ResponseStatus.OK) {
-                return result.getData();
-            } else if (retryCounter == 0) {
-                logger.debug("Error in aqicn.org, retrying once");
-                return getData(stationId, location, retryCounter + 1);
+            String error = result.getErrorMessage();
```

Comment:
Caution: result could be null

Reply:
In theory, yes. Practically, no because there is a JsonSyntaxException catch. Lets satisfy the IDE and handle it.

Reply:
Done

Reply:
Just to be clear, the case that would have triggered a NPE with your previous code is the case if response is null or an empty string. Look at description of `HttpUtil.executeUrl`:

> return the response body or <code>NULL</code> when the request went wrong

Note that you need to apply spotless to make your code building.

Reply:
Yes, you're correct - I woke up to early :)